x-pack/plugins/security_solution/public/common/components/header_page/index.tsx

```diff
@@ -71,6 +71,8 @@ interface BackOptions {
 
 export interface HeaderPageProps extends HeaderProps {
   backOptions?: BackOptions;
+  /** A component to be displayed as the back button. Used only if `backOption` is not defined */
+  backComponent?: React.ReactNode;
```

Comment:
@XavierM could you look this one over and 👍  or 👎  or 🕺 😀

Reply:
I think what you did make sense since you want to use onClick and not just href. If I understood it correctly, we should refactor all of `backOptions` to `backComponent` in the near future.

Reply:
thanks @XavierM for the comment. 
I actually think that perhaps the current `BackOptions` can be refactored to support more cases and not necessary move to always having to define a `backComponent` every time.

__Let me expand on why this was needed:__
There is actually more that we can't use with the current component other than `onClick` handler. My initial thought was to refactor the `BackOptions` and open that up to be able to control all aspects of the props given to `<LinkIcon/>`, but I think that would be too much of a lift at this time and could impact several areas of the Security Solution code. 

In our case, we need to present an action (button) that when clicked takes the user to another Kibana app (outside of Security Solutions). The current implementation of `<HeaderPage>` and `BackOptions` assumes that the "back" button will always be to another Security Solution sub-app (it uses `formatUrl()` on the `href` path passed), as well as a custom `onClick` handler that assumes the `href` path is a valid security solution react-router path.

Reply:
Yes that makes sense, I was not asking you to refactor everything in this PR